mlflow/server/js/src/experiment-tracking/components/artifact-view-components/ShowArtifactPdfView.js

```diff
@@ -0,0 +1,110 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import './ShowArtifactImageView.css';
+import { getSrc } from './ShowArtifactPage';
+import { Document, Page, pdfjs } from 'react-pdf';
+import { Pagination, Spin } from 'antd';
+import { getArtifactBytesContent } from './ShowArtifactUtils';
+import './ShowArtifactPdfView.css';
+
+class ShowArtifactPdfView extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+  state = {
+    loading: true,
+    error: undefined,
+    pdfData: undefined,
+    currentPage: 1,
+    numPages: 1,
+  };
+
+  static propTypes = {
+    runUuid: PropTypes.string.isRequired,
+    path: PropTypes.string.isRequired,
+    getArtifact: PropTypes.func,
+  };
+
+  static defaultProps = {
+    getArtifact: getArtifactBytesContent,
+  };
+
+  /** Fetches artifacts and updates component state with the result */
+  fetchPdf() {
+    const artifactLocation = getSrc(this.props.path, this.props.runUuid);
+    this.props
+      .getArtifact(artifactLocation)
+      .then((artifactPdfData) => {
+        this.setState({ pdfData: { data: artifactPdfData }, loading: false });
+      })
+      .catch((error) => {
+        this.setState({ error: error, loading: false });
+      });
+  }
+
+  componentDidMount = () => {
```

Comment:
```suggestion
  componentDidMount() {
```

I think a React lifecycle method should not be an arrow function.

Reply:
Done - thanks for the note!